test/src/e2e_vm_tests/test_programs/should_fail/vec/src/main.sw

```diff
@@ -0,0 +1,716 @@
+script;
+
+use std::{assert::assert, hash::sha256, option::Option, revert::revert, vec::Vec};
+
+struct SimpleStruct {
+    x: u32,
+    y: b256,
+}
+
+enum SimpleEnum {
+    X: (),
+    Y: b256,
+    Z: (b256,
+    b256), 
+}
+
+fn main() -> bool {
+    test_vector_new_u8();
+    test_vector_new_b256();
+    test_vector_new_struct();
+    test_vector_new_enum();
+    test_vector_new_tuple();
+    test_vector_new_string();
+    test_vector_new_array();
+    test_vector_with_capacity_u64();
+    true
+}
+
+fn test_vector_new_u8() {
+    let mut vector: Vec<u8> = ~Vec::new();
+
+    let number0 = 0u8;
+    let number1 = 1u8;
+    let number2 = 2u8;
+    let number3 = 3u8;
+    let number4 = 4u8;
+    let number5 = 5u8;
+    let number6 = 6u8;
+    let number7 = 7u8;
+    let number8 = 8u8;
+
+    assert(vector.len() == 0);
+    assert(vector.capacity() == 0);
+    assert(vector.is_empty());
+
+    vector.push(number0);
+    vector.push(number1);
+    vector.push(number2);
+    vector.push(number3);
+    vector.push(number4);
+    vector.push(false);
```

Comment:
Is it possible to reduce the size of this failing test to the bare minimum by removing all or most of the things that don't fail?

Reply:
Done.